src/Bicep.LangServer.IntegrationTests/CompletionTests.cs

```diff
@@ -1460,7 +1433,46 @@ public async Task Spread_array_completions_work_with_parentheses()
   ]
 ]
 """);
+
+        [TestMethod] // https://github.com/Azure/bicep/issues/14066
+        public Task Object_completions_work_inside_ternary() => RunCompletionTest("""
+var foo1 = {}
+var foo2 = []
+
+resource str 'Microsoft.Storage/storageAccounts@2023-04-01' = {
+  name: 'str0909'
+  properties: {
+    networkAcls: empty(foo1) ? null : {
+      defaultAction: 'Allow'
+      resourceAccessRules: empty(foo2) ? null : [
+        {
+          resourceId: ''
```

Comment:
[nit]: we don't check for completion `kind`?

Reply:
Not in these tests - they're designed to be visual and low-cost to author. What would you want to check for with `kind`?

Reply:
IIRC the `kind` is responsible for showing the small icon you see in front of each suggestion. Although, I believe these completions are being generated in a nested fashion, so it isn't necessary to test for `kind` at this level.